components/rename_channel_modal/rename_channel_modal.jsx

```diff
@@ -69,28 +113,26 @@ export class RenameChannelModal extends React.Component {
     }
 
     componentWillReceiveProps(nextProps) {
+        const {requestStatus: nextRequestStatus, serverError: nextServerError} = nextProps;
+        const {requestStatus, team} = this.props;
+
         if (!Utils.areObjectsEqual(nextProps.channel, this.props.channel)) {
             this.setState({
                 displayName: nextProps.channel.display_name,
                 channelName: nextProps.channel.name
             });
         }
-    }
-
-    shouldComponentUpdate(nextProps, nextState) {
-        if (!nextProps.show && !this.props.show) {
-            return false;
-        }
 
-        if (!Utils.areObjectsEqual(nextState, this.state)) {
-            return true;
+        if (requestStatus !== nextRequestStatus && nextRequestStatus === RequestStatus.SUCCESS) {
+            this.handleHide();
+            browserHistory.push('/' + team.name + '/channels/' + this.state.channelName);
         }
 
-        if (!Utils.areObjectsEqual(nextProps, this.props)) {
-            return true;
+        if (requestStatus !== nextRequestStatus && nextRequestStatus === RequestStatus.FAILURE) {
+            this.setError(nextServerError);
+        } else {
+            this.unsetError();
```

Comment:
Instead of using this success/error flow, you can actually just await on the action in the handleSubmit function, like so:

```
const {data, error} = await updateChannel(channel);
// handle success if data is set, handle error if error is set
```

That way we don't need to pass the request status in as a prop and we reduce the complexity bit. This is a relatively recent change to how actions work in mattermost-redux work

Reply:
Ah I see, seems like a nice and sophisticated way to do it. I'll start implementing it right away!